test_functests_server.py

```diff
@@ -0,0 +1,129 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+import server
+import socket
+import pytest
+import time
+from multiprocessing import Process
+
+
+addr = ("127.0.0.1", 8000)
+
+
+@pytest.yield_fixture(scope='module', autouse=True)
+def server_setup():
+    process = Process(target=server.run_server)
+    process.daemon = True
+    process.start()
+    time.sleep(0.1)
+    yield process
+
+
+@pytest.fixture(scope='function')
+def client_setup():
+    client = socket.socket(
+        socket.AF_INET, socket.SOCK_STREAM, socket.IPPROTO_IP
+    )
+    client.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+    client.connect(addr)
+    return client
+
+
+################
+# FUNCTIONAL TESTS
+################
+
+
+def test_client_receives_ok_on_image_request(client_setup):
+    client = client_setup
+    client.sendall(
+        b"GET /images/sample_1.png HTTP/1.1\r\n"
+        b"Host: www.host.com:80\r\n"
+        b"\r\n"
+    )
+    ok_header = b"HTTP/1.1 200 OK"
+    server_response = client.recv(4096)
+    client.close()
+    assert ok_header in server_response
```

Comment:
I'd like to see a more thorough testing of the response here.  is the content length header correct?  Is the content type header correct?  Does the proper separator come between the head and the body of the response?